### DISCUSSION TITLE
What's a Compound Button?

### DISCUSSION THREAD

**Discussion by mtanco** [h2oai/wave#1259] [Q&A]
The documentation for the `caption` parameter of `ui.button` states that using the caption make a compound button: https://github.com/h2oai/wave/blob/980f0dcf3468e289e3fedcd74767ace8cf6bc517/ui/src/button.tsx#L47
`Compound Button` is never mentioned in our docs, and not really clear from the source code what this would mean. A google search implies compound buttons have 2 states, clicked and not clicked, but I am seeing no difference in behavior between a button with and without a caption after its clicked. 

```python
from h2o_wave import main, app, Q, ui


@app('/')
async def serve(q: Q):

    if not q.app.initialized:
        q.page["form"] = ui.form_card(
            box="4 1 4 4",
            items=[
                ui.button(name="button1", label="Compound Button", caption="This is some text"),
                ui.button(name="button2", label="Normal button"),
                # ui.link(label="Download Data", path=q.app.data_link, download=True, button=True)
            ]
        )
        q.app.initialized = True

    elif q.args.button1:

        q.page["form2"] = ui.form_card(
            box="4 5 4 4",
            items=[
               ui.text("you clicked button one")
            ]
        )

    elif q.args.button2:

        q.page["form2"] = ui.form_card(
            box="4 5 4 4",
            items=[
               ui.text("you clicked button two")
            ]
        )

    await q.page.save()
```

**Comment by mturoci** [comment]
The term "compound" comes from [FluentUI](https://developer.microsoft.com/en-us/fluentui#/controls/web/button) and is just trying to say that the button will be made up of 2 lines of text in contrast to a typical single line. Nothing special apart from that.

**Reply by mtanco** [reply]
I would suggest removing this from the the parameter description, it's very low priority but I think it can be quite confusing to use a word we never define, especially since google shows a different definition (android's compoundButton of checked and unchecked state)

**Reply by mturoci** [reply]
Agree, will remove it.

**Reply by mturoci** [reply]
Done in - https://github.com/h2oai/wave/commit/e5e9cc8a29b55451493f613f10aa06c780e4cea1
